recipes/automake/all/conanfile.py

```diff
@@ -34,107 +32,91 @@ def configure(self):
-        self.env_info.AUTOMAKE_DATADIR = automake_datadir
+            # tracing using m4 on Windows returns Windows paths => use cygpath to convert to unix paths
+            replace_in_file(self, path.join(self.source_path, "bin", "aclocal.in"),
+                            "          $map_traced_defs{$arg1} = $file;",
+                            "          $file = `cygpath -u $file`;\n"
+                            "          $file =~ s/^\\s+|\\s+$//g;\n"
+                            "          $map_traced_defs{$arg1} = $file;")
 
-        automake_libdir = tools.unix_path(self._automake_libdir)
-        self.output.info("Setting AUTOMAKE_LIBDIR to {}".format(automake_libdir))
-        self.env_info.AUTOMAKE_LIBDIR = automake_libdir
+        _ = self._autotools()
 
-        automake_perllibdir = tools.unix_path(self._automake_libdir)
-        self.output.info("Setting AUTOMAKE_PERLLIBDIR to {}".format(automake_perllibdir))
-        self.env_info.AUTOMAKE_PERLLIBDIR = automake_perllibdir
+    def package(self):
+        autotools = self._autotools()
+        # KB-H013 we're packaging an application, place everything under bin
+        autotools.install(args=[f"DESTDIR={unix_path(self, path.join(self.package_folder, 'bin'))}"])
 
-        automake = tools.unix_path(os.path.join(self.package_folder, "bin", "automake" + bin_ext))
-        self.output.info("Setting AUTOMAKE to {}".format(automake))
-        self.env_info.AUTOMAKE = automake
+        copy(self, "COPYING*", src=self.source_folder, dst=path.join(self.package_folder, "licenses"))
+        rmdir(self, path.join(self._datarootdir, "info"))
+        rmdir(self, path.join(self._datarootdir, "man"))
+        rmdir(self, path.join(self._datarootdir, "doc"))
 
-        self.output.info("Append M4 include directories to AUTOMAKE_CONAN_INCLUDES environment variable")
+    def _set_env(self, var_name, var_path):
+        self.output.info(f"Setting {var_name} to {var_path}")
+        self.buildenv_info.define_path(var_name, var_path)
+        setattr(self.env_info, var_name, var_path)
 
-        self.user_info.compile = os.path.join(self._automake_libdir, "compile")
-        self.user_info.ar_lib = os.path.join(self._automake_libdir, "ar-lib")
+    def package_info(self):
+        # KB-H013 we're packaging an application, hence the nested bin
+        bin_dir = path.join(self.package_folder, "bin", "bin")
+        self.output.info(f"Appending PATH environment variable:: {bin_dir}")
+        self.buildenv_info.prepend_path("PATH", bin_dir)
+        self.env_info.PATH.append(bin_dir)
+
+        for var in [("ACLOCAL", path.join(self.package_folder, "bin", "aclocal")),
+                    ("AUTOMAKE_DATADIR", self._datarootdir),
+                    ("AUTOMAKE_LIBDIR", self._automake_libdir),
+                    ("AUTOMAKE_PERLLIBDIR", self._automake_libdir),
+                    ("AUTOMAKE", path.join(self.package_folder, "bin", "automake"))]:
+            self._set_env(*var)
```

Comment:
No, your function _set_env always relies on `define_path`, it's not what we want for ACLOCAL.

Reply:
I have limited experience with Autotools in general except for my current attempts to make these recipes work on Windows with msys2.

Could you elaborate a bit why ACLOCAL should not use define_path?

Isn't it just the path to a single binary?